### ISSUE TITLE
Create new endpoint to manage API configuration

### ISSUE THREAD

**Issue by crd1985** [wazuh/wazuh#4822]
# Description

Currently, the only way to change any API setting is to edit the `api.yaml` file. This method forces users to login to the deployed server or use the `configure_api.sh` script. The goal of this issue is to extend API capabilities to handle its own configuration so everything could be done via REST requests.

Most of settings may be hot-changed without restarting the API daemon. They are quite unlikely to make the API service inaccessible, so we could call them 'soft settings':
- `behind_proxy_server`
- `rbac`.`mode`
- `logs`.`level`
- `cors`
- `cache`
- `use_only_authd`
- `experimental_features`

The rest of settings may need a restart of the service and cannot guarantee that the service is working again, so we could call them 'hard settings'. The only way to change this settings will be editing the `api.yaml` file. The reason is to ensure that if an administrator makes a mistake and breaks the environment, there is another chance to restore it:
- `host`
- `port`
- `https`
- `logs`.`path`
- `drop_privileges`

## Design

The following endpoints will be created to handle soft settings:
- Read access to configuration:
   - `GET /manager/api/config`
   - `GET /cluster/api/config`
- Selectively edit configuration:
   - `PUT /manager/api/config`
   - `PUT /cluster/api/config`
- Restore configuration to factory settings:
   - `DELETE /manager/api/config`
   - `DELETE /cluster/api/config`

Some use cases are:
- Get configuration
```json
# GET /manager/api/config
{
        "behind_proxy_server": false,
        "rbac": {
            "mode": "black"
        },
        "logs": {
            "level": "info"
        },
        "cors": true,
        "cache": {
            "enabled": false,
            "debug": false,
            "time": 0.750
        },
        "use_only_authd": false,
        "experimental_features": false
}
```
- Change setting
```json
# PUT /manager/api/config
{"cache": 
    {"enabled": false}
}
```
```json
# Response 200
{"message": "API configuration update successfully."
}
```
- Restore configuration:
```json
# DELETE /manager/api/config
{"message": "API configuration restored to factory settings."
}
```

Basically, the logic of these endpoints is to modify the `api.yaml` file in `/var/ossec/api/configuration`. The spec should be detailed enough to validate the body of the PUT endpoints that fits the configuration template perfectly.

### RBAC

The resource here is the already implemented `node:id:{node_id}`. See `manager.py` for a more detailed reference. However, new actions should be created for read, modifying and restoring the configuration.

## Tasks
- [x] Modify `spec.yaml` to include new endpoints
- [x] Add new functions in the framework to give support to the required logic and RBAC
- [x] Update RBAC catalogue
- [x] Review default roles
- [x] Write integration tests (basic, RBAC while-list and RBAC black-list)
